py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,463 @@
+                squeze_model='/global/homes/e/ebie/local/SQUEzE/data/BOSS_train_64plates_model.json',
+                qnp_lines=None, qnp_lines_bal=None):
+    """This will create a single zcat file from a set of user inputs.
+    
+    Parameters
+    ----------
+    tile : :class:'str'
+        The TILEID of the tile to process.
+    night : :class:'str'
+        The date associated with the observation of the 'tile' used.
+        * Must be in YYYYMMDD format
+    petal_num : :class:'int'
+        If 'all_petals' isn't used, the single petal to create a zcat for.
+    zcatdir : :class:'str'
+        The location for the daily redrock output.
+    outputdir : :class:'str'
+        The filepath to write out the zcat file.
+    qn_flag : :class:'bool'
+        Flag to add QuasarNP data (or not) to the zcat file.
+    sq_flag : :class:'bool'
+        Flag to add SQUEzE data (or not) to the zcat file.
+    abs_flag : :class:'bool'
+        Flag to add MgII Absorption data (or not) to the zcat file.
+    qnp_model : :class:'str' or 'h5 array'
+        The filename and path for the QuasarNP model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+    squeze_model : :class:'str' or 'numpy.array'
+        The filename and path for the SQUEzE model file. IF this function
+        is run as part of a loop, the calling function will load the model
+        file and this will be an array instead.
+    qnp_lines : :class:'list' or None
+        The list of lines to use in the QuasarNP model to test against. If
+        the script is run in loop mode, the list is passed from the calling
+        function, otherwise it's created below.
+    qnp_lines : :class:'list' or None
+        The list of BAL lines to use for QuasarNP to identify BALs. If
+        the script is run in loop mode, the list is passed from the calling
+        function, otherwise it's created below.
+        
+    Outputs
+    -------
+    A FITS catalog that incorporates redrock, QuasarNP, SQUEzE, and MgII
+    absorption redshifts and confidence values.
+    
+    Notes
+    -----
+    - There will be many here.
+    """
+    # EBL Load the SQUEzE Model file first. This is a very large file,
+    # so if multiple petals are to be processed, we only want to load
+    # it into memory once.
+    if isintance(qnp_model, str) and qn_flag:
+        tmark('    Loading QuasarNP Model file')
+        qnp_lines = ['LYA', 'CIV(1548)', 'CIII(1909)', 'MgII(2796)', 'Hbeta', 'Halpha']
+        qnp_lines_bal = ['CIV(1548)']
+        qnp_model = load_model('/global/cfs/cdirs/desi/science/lya/qn_models/boss_dr12/qn_train_coadd_indtrain_0_0_boss10.h5')
+        tmark('      QNP model file loaded')
+    if isinstance(squeze_model, str) and sq_flag:
+        tmark('    Loading SQUEzE Model file')
+        squeze_model = Model.from_json(load_json(squeze_model))
```

Comment:
Let's create a separate `load_sq_model()` function instead of this complicated construction. In fact let's:

- Have separate functions for `load_sq_model(filename=X)` and `load_qn_model(filename=Y)`, where `X` and `Y` default to the standard locations at NERSC.
- Write `create_zcat` to always take the loaded models as inputs rather than either a string or a loaded model.

Loading the SQUEzE model is likely to be a critical bottleneck, so let's have single, transparent functions to load the models and not embed the model-loading in the `create_zcat()` function, even if that loading is just an option.

Reply:
@geordie666 Model filenames are generated by separate functions now and use environment variables. Model files are loaded by their own convenience functions.